completed / выполнено

drsispb opened the pull request.

drsispb committed:
completed / выполнено

alexandermatseev commented on skill_box_15/15.2.py:
```diff
@@ -0,0 +1,22 @@
+        new_list_first_day.append(i)
+    else:
+        new_list_second_day.append(i)
+    count += 1
```
@drsispb очень странно использовать еще один счетчик, если у тебя используется цикл с внутренним счетчиком.
как вариант посмотри в сторону [enumirate](https://proglib.io/p/python-enumerate-uproshchaem-cikly-s-pomoshchyu-schetchikov-2020-12-08), другой вариант пройтись не по списку, а по длине списка, тогда будут учитываться индексы (что тебе и нужно)

alexandermatseev commented on skill_box_15/15.3.py:
```diff
@@ -0,0 +1,29 @@
+    if int(i) > effectiv:
+        bad_effectiv.append(effectiv)
+    print('Эффективность', count, 'клетки:', effectiv)
+    count += 1
```
@drsispb Счетчик только лишь для вывода информации я думаю лишний, можно использовать i?

alexandermatseev commented on skill_box_15/15.3.py:
```diff
@@ -0,0 +1,29 @@
+bad_effectiv = []
+for i in list:
+    effectiv = int(input('Введите эффективность '))
+    if int(i) > effectiv:
```
@drsispb i разве и так не является числом?

alexandermatseev commented on skill_box_15/15.4.py:
```diff
@@ -0,0 +1,36 @@
+old_list = []
+count = 0
+
+while count != N:
```
@drsispb прекрасно что ты работаешь с разным типом циклов, но тут более уместен цикл for как мне кажется, и счетчик count будет не нужен

alexandermatseev commented on skill_box_15/15.4.py:
```diff
@@ -0,0 +1,36 @@
+    card = input(' Видеокарта: ')
+    old_list.append(card)
+check = 0
+for i in old_list:
```
@drsispb я так понимаю это сортировка? посмотри в сторону [sorted](https://pyneng.readthedocs.io/ru/latest/book/10_useful_functions/sorted.html) или list.sort()

alexandermatseev commented on skill_box_15/15.4.py:
```diff
@@ -0,0 +1,36 @@
+    if int(check) < int(i):
+        check = i
+new_list = []
+for _ in old_list:
```
@drsispb 1) переменной цикла "_" называют переменную, которая в цикле не используется, у тебя она используется на протяжении всего цикла и не однократно. 2) Так как ты производишь сортировку выше, достаточно удалить последний элемент из списка и вывести его как новый. Не зачем еще раз проходить по списку явно.

alexandermatseev commented on skill_box_15/15.5.py:
```diff
@@ -0,0 +1,29 @@
+choice_film = int(input('Сколько фильмов хотите добавить? '))
+count = 0
+total_film = len(films)
+while count != choice_film:
```
@drsispb Нужно поработать над выбором применяемости циклов, тут удобнее также использовать for и избавится от счетчика.

alexandermatseev commented on skill_box_15/15.5.py:
```diff
@@ -0,0 +1,29 @@
+while count != choice_film:
+    count += 1
+    film = str(input('Введите название фильма: '))
+    if films.count(film) > 0:
```
@drsispb Смысл задания ты понял верно, и решение рабочее, но более логично использовать условие вхождения введеного фильма в список film in film_list примерно так. Плюс в задании этого нет, но стоит предусмотреть защиту от регистра введенного фильма, стоит перевести при сравнении названий все к одному регистру. а то у тебя будет два фильма в списке Терминатор и терминатор и ТеРмИнАтОр

alexandermatseev commented on skill_box_15/15.6.py:
```diff
@@ -0,0 +1,25 @@
+Количество уникальных букв: 2'''
+
+
+word = input('Введите слово: ')
```
@drsispb лайфхак для перевода из строки в список input().split(), попробуй и [почитай](https://pythonz.net/references/named/str.split/).

alexandermatseev commented on skill_box_15/15.6.py:
```diff
@@ -0,0 +1,25 @@
+long_word = len(word_list)
+count = 0
+for i in word_list:
+    for x in word_list:
```
@drsispb мне кажется использование двух списков избыточно. Ранее ты использовал метод count, мне кажется он сюда бы смог подойти. [Прочти](https://pythonstart.ru/list/count-python).
Он резко сократит твой код.

alexandermatseev commented on skill_box_15/15.7.py:
```diff
@@ -0,0 +1,44 @@
+N = int(input('Количество контейнеров: '))
+contener_list = []
+count = 0
+while count != N:
```
@drsispb цикл for и никакого счетчика.И может стоит тут применить функцию для контроля ввода отдельно в цикле?

alexandermatseev commented on skill_box_15/15.8.py:
```diff
@@ -0,0 +1,26 @@
+half_s = []
+long = len(spisok)
+step = int(input('Введите сдвиг: '))
+for i in spisok[:long - step]:
```
@drsispb попробуй одним циклом решить, дам подсказку, циклом ты должен проходится не по списку а по длине, а вот в цикле работать со списком и сдвигом по какому то условию. Для сдвига есть определенная закономерность, ее нужно понять.

alexandermatseev commented on skill_box_15/15.9.py:
```diff
@@ -0,0 +1,23 @@
+
+word = str(input('Введите слово: '))
+un_word = ''
+for a in word[::-1]:
```
@drsispb Красота языка в том, что тут тебе даже не нужен цикл, тут ты уже сделал всю работу и теперь тебе всего лишь нужно сравнить с оригиналом

alexandermatseev commented on skill_box_15/15.10.py:
```diff
@@ -0,0 +1,32 @@
+# '''спасибо интернету'''
+# star_list = [1, 4, -3, 0, 10]
+# star_list.sort()
+# print(star_list)
```
Это самый верный вариант.

alexandermatseev commented on skill_box_15/15.10.py:
```diff
@@ -0,0 +1,32 @@
+# for i in star_list:
+#     if i <= 0:
+'''вообще тоже интернет, но пытался сам '''
+def insertion_sort(nums):
```
@drsispb Прекрасно. Текстом назови мне самую быструю сортировку списка и приведи код)

drsispb replied:
sort т.к. его сложность O (n log(N))

drsispb committed:
убрал счетчик, исправил прохождение цикла по позициям.
в новый список дабвил присвоение по индексу.

drsispb committed:
убрал счетчик, использовал i для вывода информации

drsispb committed:
убрал присвоение int к i
(изначально вод был иной, осталось с прошлой попытки)

drsispb committed:
заменил цикл while на for в строке 21, за счет чего убрал count.
убрал проверку за счет сортировки списка. стр 25
Исправил присвоение, взяв за условие что крайние значение списка будет наибольшим.
Во втором цикле присвоил имя переменной.

drsispb committed:
заменил цикл while на for и убрал счетчик
в строке 23 сделал привидение к ригистрам, согласно списка
в строке 25 исправил проверку вхождения в список

drsispb committed:
заменил цикл while на for и убрал счетчик
в строке 23 сделал привидение к ригистрам, согласно списка
в строке 25 исправил проверку вхождения в список

drsispb committed:
заменил цикл while на for и убрал счетчик
в строке 23 сделал привидение к ригистрам, согласно списка
в строке 25 исправил проверку вхождения в список

drsispb committed:
поменял способ проверки на count
убрал создание лишнего списка

drsispb committed:
применил способ split и сделал проверку напрмую в список

drsispb committed:
добавил функцию проверки, заменил цикл while на for.
да кого я обманываю, просто твой код под свои аргументы переписал)

drsispb committed:
убрал лишний цикл, сделал проход по длинне

drsispb committed:
убрал цикл, просто развернул строку

alexandermatseev commented on skill_box_15/15.4.py:
```diff
@@ -17,20 +17,15 @@
-        new_list.append(_)
+for m in old_list:
+    if m != old_list[N - 1]:
+        new_list.append(m)
```
@drsispb   по моему можно проще, сделай копию старого списка  и удали в копии последний элемент, по моему это отработает как надо и второй цикл вообще не понадобится

alexandermatseev commented on skill_box_15/15.5.py:
```diff
@@ -17,12 +17,12 @@
+for _ in range(choice_film):
     film = str(input('Введите название фильма: '))
-    if films.count(film) > 0:
+    film = film[:1].upper() + film[1:].lower()
```
@drsispb можно пройтись по списку films и каждый элемент привести к примеру к нижнему регистру.
и вводимый файл привести к регистру нижнему: input('Введите название фильма: ').lower()

alexandermatseev commented on skill_box_15/15.5.py:
```diff
@@ -17,12 +17,12 @@
-while count != choice_film:
-    count += 1
+for _ in range(choice_film):
     film = str(input('Введите название фильма: '))
```
@drsispb не нужно приводить к str инпут всегда принимает строку

alexandermatseev commented on skill_box_15/15.8.py:
```diff
@@ -14,13 +14,13 @@
\ No newline at end of file
+
+for _ in range(long):
+    a = spisok[_] - step - 1
```
@drsispb задай имя переменной цикла если ты ее используешь

drsispb committed:
исправил недочеты.
в 15.5 сделал приведение к общему регистру

drsispb committed:
15.10 ничего не делал
